dds/idl/dds_generator.h

```diff
@@ -725,8 +731,8 @@ void generateCaseBody(
       rhs = use_cxx11 ? "tmp" : "tmp.out()";
     } else if (use_cxx11 && (br_cls & (CL_ARRAY | CL_SEQUENCE))) {  //array or seq C++11
       rhs = "IDL::DistinctType<" + brType + ", "
-        + dds_generator::scoped_helper(branch->field_type()->name(), "_")
-        + "_tag>(tmp)";
+        + dds_generator::get_tag_name(dds_generator::scoped_helper(branch->field_type()->name(), "_"), false)
```

Comment:
```suggestion
        + dds_generator::get_tag_name(dds_generator::scoped_helper(branch->field_type()->name(), "_"))
```

Reply:
Also it's kinda weird to me to be using `scoped_helper` and `get_tag_name` together.

Reply:
Each time scoped_helper was called was as a consistent way to convert from a scoped name to a string. 

Because of the changes to get_tag_name it makes a difference what state the string is passed to it.  Basically if the passed strings contain different :: or _ it would map differently.